pyscf/pbc/dft/test/test_multigrid2.py

```diff
@@ -62,33 +91,145 @@ def _multigrid2_energy_grad(cell, xc, spin=0):
+
+    def test_orth_lda_veff(self):
         xc = 'lda, vwn'
-        e0, g0 = _fftdf_energy_grad(cell, xc)
-        e,  g  = _multigrid2_energy_grad(cell, xc, 0)
-        e1, g1 = _multigrid2_energy_grad(cell, xc, 1)
-        assert abs(e-e0) < 1e-8
-        assert abs(e1-e0) < 1e-8
-        assert abs(g-g0).max() < 2e-5
-        assert abs(g1-g0).max() < 2e-5
-
-    def test_orth_gga(self):
+        _test_veff(cell_orth, xc, dm, spin=0)
+        _test_veff(cell_orth, xc, dm1, spin=1)
+
+    def test_orth_gga_veff(self):
         xc = 'pbe, pbe'
-        e0, g0 = _fftdf_energy_grad(cell, xc)
-        e,  g  = _multigrid2_energy_grad(cell, xc, 0)
-        e1, g1 = _multigrid2_energy_grad(cell, xc, 1)
+        _test_veff(cell_orth, xc, dm, spin=0)
+        _test_veff(cell_orth, xc, dm1, spin=1)
+
+    def test_nonorth_lda_veff(self):
+        xc = 'lda, vwn'
+        _test_veff(cell_nonorth, xc, dm, spin=0)
+        _test_veff(cell_nonorth, xc, dm1, spin=1)
+
+    def test_nonorth_gga_veff(self):
+        xc = 'pbe, pbe'
+        _test_veff(cell_nonorth, xc, dm, spin=0)
+        _test_veff(cell_nonorth, xc, dm1, spin=1)
+
+    def test_orth_lda_dft(self):
+        xc = 'lda, vwn'
+        e0, g0 = _fftdf_energy_grad(He_orth, xc)
+        e,  g  = _multigrid2_energy_grad(He_orth, xc, 0)
+        e1, g1 = _multigrid2_energy_grad(He_orth, xc, 1)
+        assert abs(e-e0) < 1e-6
+        assert abs(e1-e0) < 1e-6
+        assert abs(g-g0).max() < 1e-6
+        assert abs(g1-g0).max() < 1e-6
+
+    def test_orth_gga_dft(self):
+        xc = 'pbe, pbe'
+        e0, g0 = _fftdf_energy_grad(He_orth, xc)
+        e,  g  = _multigrid2_energy_grad(He_orth, xc, 0)
+        e1, g1 = _multigrid2_energy_grad(He_orth, xc, 1)
         assert abs(e-e0) < 1e-6
         assert abs(e1-e0) < 1e-6
-        assert abs(g-g0).max() < 1e-4
-        assert abs(g1-g0).max() < 1e-4
+        assert abs(g-g0).max() < 1e-6
+        assert abs(g1-g0).max() < 1e-6
+
+    def test_nonorth_lda_dft(self):
+        xc = 'lda, vwn'
+        e0, g0 = _fftdf_energy_grad(He_nonorth, xc)
+        e,  g  = _multigrid2_energy_grad(He_nonorth, xc, 0)
+        e1, g1 = _multigrid2_energy_grad(He_nonorth, xc, 1)
+        assert abs(e-e0) < 1e-7
+        assert abs(e1-e0) < 1e-7
+        assert abs(g-g0).max() < 1e-7
+        assert abs(g1-g0).max() < 1e-7
+
+    def test_nonorth_gga_dft(self):
+        xc = 'pbe, pbe'
+        e0, g0 = _fftdf_energy_grad(He_nonorth, xc)
+        e,  g  = _multigrid2_energy_grad(He_nonorth, xc, 0)
+        e1, g1 = _multigrid2_energy_grad(He_nonorth, xc, 1)
+        assert abs(e-e0) < 1e-7
+        assert abs(e1-e0) < 1e-7
+        assert abs(g-g0).max() < 1e-7
+        assert abs(g1-g0).max() < 1e-7
+
+    def test_orth_j_dft(self):
+        xc = ''
+        e0, g0 = _fftdf_energy_grad(He_orth, xc)
+        e,  g  = _multigrid2_energy_grad(He_orth, xc, 0)
+        e1, g1 = _multigrid2_energy_grad(He_orth, xc, 1)
+        assert abs(e-e0) < 1e-6
```

Comment:
If cell.precision=1e-8, 1e-6 error seems too big. I observed that in certain scenario, the reference provided by the default fftdf is not accurate enough, with errors slightly larger than the required cell.precision. Is this the case here?